src/monattack.cpp

```diff
@@ -2716,9 +2716,8 @@ bool mattack::grab( monster *z )
             target->add_msg_player_or_npc( m_info, tech.avatar_message.translated(),
                                            tech.npc_message.translated(), z->name() );
         } else {
-            target->add_msg_player_or_npc( m_info, _( "The %s tries to grab you, but you break its grab!" ),
-                                           _( "The %s tries to grab <npcname>, but they break its grab!" ),
-                                           z->name() );
+            add_msg_if_player_sees( *z, m_info, _( "The %1$s tries to grab %2$s, but %2$s break its grab!" ),
```

Comment:
```suggestion
            add_msg_if_player_sees( *z, m_info, _( "The %1$s tries to grab %2$s, but %2$s breaks its grab!" ),
```
This assumes that it's an NPC. If not, it should still be `_( "The %s tries to grab you, but you break its grab!" )`.

Reply:
This message will fire both on player and on NPC, and all other monster's targets. Unfortunately there isn't a version of `add_msg_if_player_sees` which can separate accept parameters for player and for NPCs, like `add_msg_player_or_npc`, so we need some universal phrase which would fit in both situations.

Reply:
> This message will fire both on player and on NPC, and all other monster's targets. Unfortunately there isn't a version of `add_msg_if_player_sees` which can separate accept parameters for player and for NPCs, like `add_msg_player_or_npc`, so we need some universal phrase which would fit in both situations.

Or check whether `target` is the player.